src/hotspot/share/oops/weakHandle.hpp

```diff
@@ -54,6 +54,7 @@ class WeakHandle {
   inline oop peek() const;
   void release(OopStorage* storage) const;
   bool is_null() const { return _obj == nullptr; }
+  void set_null() { _obj = nullptr; }
```

Comment:
I would be a lot happier if WeakHandle.release() was non-const and set the _obj to nullptr, so that nothing can do this without release the weak handle.
This could be a follow-up RFE though since it affects other WeakHandle.release() calls.

Reply:
Just to be clear, the weak handle is released, but the `_obj` field in the WeakHandle is not
cleared so it is a dangling oop reference.

I didn't want to change `release(OopStorage* storage)` because that had a larger fan out
of changes outside the scope of what I was trying to accomplish.